### PR TITLE
Enhancement: Update User Activity Query

### PR DESCRIPTION
**Pull request by FahadKhalid210**
update user activity query to improve average time spent in course

### Description:
In our ongoing analysis of **PT** data, we've identified discrepancies in the time spent by users in courses. Additionally, relying solely on "browser" events for these calculations has proven insufficient. After thorough investigation, we've determined that extending the time span from 10 seconds to 120 seconds and selecting all events where the course ID is not null provides a more accurate representation of the time users spend in courses. 

Therefore, this pull request updates the query to enhance the accuracy of the average time spent in courses.

**Before:** 
image

**After:**
image

### CODE REVIEW

File: tutorcairn/templates/cairn/apps/superset/bootstrap/courseoverview.json [hunk 1]
```diff
@@ -1909,7 +1909,7 @@
                 "offset": 0,
                 "params": "{\"remote_id\": 49, \"database_name\": \"admin\"}",
                 "schema": "openedx",
-                "sql": "SELECT user_id,\n  course_id,\n  time,\n  arrayJoin(range(floor(toUInt64(toUnixTimestamp(time) - 5)), ceil(toUInt64(toUnixTimestamp(time) + 5)))) AS timestamp\nFROM events\nWHERE event_source = 'browser'",
+                "sql": "SELECT user_id,\n  course_id,\n  time,\n  arrayJoin(range(floor(toUInt64(toUnixTimestamp(time))), ceil(toUInt64(toUnixTimestamp(time) + 120)))) AS timestamp\nFROM events\nWHERE course_id <> '' and user_id > 0",
```
Comment (DawoudSheraz) [inline tutorcairn/templates/cairn/apps/superset/bootstrap/courseoverview.json:1912]: So we are ignoring browser entirely in this new query? Why is that? The timestamp change is understandable. 
Also, if we can add some visualizations (before/after), it would make it easier to understand the impact.

Comment (FahadKhalid210) [reply tutorcairn/templates/cairn/apps/superset/bootstrap/courseoverview.json:1912]: We are now picking both events(browser and server) where we have course_id and user_id because some events have source "server"(e.g `edx.grades.problem.submitted`) which we missed previously.

File: changelog.d/20240405_151230_fahad.khalid.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- [Improvement] Update User Activity dataset query to improve average time spent in course. (by @Fahadkhalid210)
```
Comment (DawoudSheraz) [inline changelog.d/20240405_151230_fahad.khalid.md]: nit: we can add context here on why this change provides a better data then before.

**Review by DawoudSheraz — APPROVED** [review]
